### PR TITLE
Created codeql.yml for code scanning

### PR DESCRIPTION
**Pull request by or-akl**
Hey Team,

I've created a YAML file as part of GitHub's security tools to scan for misconfigured patterns in our code. This will help enhance our code security.
Thanks!

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "or-appsec" ]
```
Comment (michelsk) [inline .github/workflows/codeql.yml:5]: is this the main branch?

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
+    permissions:
+      security-events: write
+      actions: read
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'csharp' ]
```
Comment (michelsk) [inline .github/workflows/codeql.yml:24]: why are you using matrix if you are using 1 item each time? in all the PR's...

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  pull_request:
+    branches: [ "or-appsec" ]
+  schedule:
+    - cron: '25 14 * * 4'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
```
Comment (michelsk) [inline .github/workflows/codeql.yml:15]: this will always default to 360 as there no swift in matrix

**Review by michelsk — COMMENTED** [review]
if the scan steps are the same for all lang's, it might be better to create a reusable or composite action and call it each time instead of duplicating

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (or-akeyless) [reply .github/workflows/codeql.yml:5]: this should be main/master, yes

**Review by OriBenHur-akeyless — COMMENTED** [review]
As I've said please use a reusable instead of repeating the same code over and over

**Review by AkeylessRoy — APPROVED** [review]

**Pull request merged**